userland/linker.ld

```diff
@@ -2,7 +2,7 @@ ENTRY(_start)
 
 MEMORY {
     FLASH (rx) : ORIGIN = 0x80000000, LENGTH = 0x00080000
-    SRAM (RWX) : ORIGIN = 0x00000000, LENGTH = 0x00002000
+    SRAM (RWX) : ORIGIN = 0x00000000, LENGTH = 0x0008000
```

Comment:
We want to support multiple simultaneous processes, in general, so while none of the anticipated chips are 8KB, each process's RAM is currently 8K.

This is currently also hardcoded in the platform definition in the kernel, so just changing this won't actually work.

Is there a good reason to support much more than 8KB for this particular pull-request? I.e., do C++ applications require more than 8K in general?